Development.md

```diff
@@ -0,0 +1,263 @@
+
+$ convox exec $WEB_PID env --app $STACK_NAME > .env
+```
+
+Now you have a bunch of secrets that will let your laptop interact with AWS APIs:
+
+```
+$ cat .env
+CLIENT_ID=noah@convox.com
+RACK=dev
+SUBNETS=subnet-13de3139,subnet-b5578fc3,subnet-21c13379
+AWS_ACCESS=AKIAIOSFODNN7EXAMPLE
+AWS_SECRET=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+AWS_REGION=us-east-1
+DYNAMO_RELEASES=dev-releases
+CLUSTER=dev-Cluster-1E4XJ0PQWNAYS
+PASSWORD=45e0f109-3f56-4b30-9b5a-b0939b8a4c25
+...
+```
+
+Note: `convox install --development` is deprecated in favor of introspecting with `convox env`.
+
+## Development Rack Docker VM
+
+A local development rack is setup with `convox start` which requires a working Docker environment. To setup a Docker environment, see the  [Docker Machine](https://docs.docker.com/machine/) docs. You can then run the project:
+
+```
+$ docker-machine start default
+$ eval $(docker-machine env default)
+
+$ convox start
+RUNNING: docker build -t convox-start-icytafnqqb /Users/noah/go/src/github.com/convox/rack
+web      | running: docker run -i --name rack-web...
+web      | [negroni] listening on :3000
+```
+
+Now you can log into the development Rack API and interact with your Convox resources:
+
+```
+$ convox login $(docker-machine ip default)
+$ convox instances
+ID          AGENT  STATUS  STARTED      PS  CPU    MEM   
+i-6cf228f7  on     active  2 hours ago  4   0.00%  10.42%
+i-146c2f97  on     active  2 hours ago  1   0.00%  3.21% 
+i-c7de605c  on     active  2 hours ago  0   0.00%  0.00% 
+```
+
+## Golang Unit Test Suite
+
+Rack has a suite of Golang unit and integration tests that offer very fast feedback (< 1 minute) about system correctness. They require a running Docker environment.
+
+```
+$ make test
+docker info >/dev/null
+go get -t ./...
+env PROVIDER=test go test -v -cover ./...
+=== RUN   TestGitUrl
+=== RUN   TestDockerCompose
+=== RUN   TestEnvFile
+...
+
+$ echo $?
+0
+```
+
+GitHub and Travis CI are configured to require that tests are passing before PR can be merged.
+
+The most complex tests setup a stub AWS and Docker httptest web servers to simulate various request and response cycles. This can be challening to write but represent a very powerful way to verify Convox behavior.
+
+## API Changes
+
+A common and imple thing to do is to fix a bug or make an enhancement to the Rack APIs. For example, maybe the `GET /system` endpoint would be more helpful if it included the ELB hostname, so you'd like to add this.
+
+The `convox/rack/api` package has a few key concepts:
+
+* Swagger Manifest (rack/api/manifest.yml). Defines all API endpoints and responses
+* Golang Client (rack/client). Bindings that talk to the HTTP API and returns Golang structs, slices and errors
+* CLI (rack/cmd/convox). High level tool that lets developers issue commands like `convox/deploy`.
+* Routes (rack/api/controllers/routes). A `gorilla/mux` configuration of request URL patterns and HTTP verbs and what handlers they go to
```

Comment:
Might use commas here instead:

`configuration of request URL patterns, HTTP verbs, and what handlers they go to`